sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -36,4 +36,9 @@ private[jdbc] class JDBCOptions(
   val upperBound = parameters.getOrElse("upperBound", null)
```

Comment:
I am fine with changing this to require as the short-circuit should occur at the very top instead of as the processing moves through. `sys.error` was how the old code was written, so I just had kept it as is.